### PR TITLE
bpo-37285: setup.py: avoid joining the SDKROOT twice

### PR DESCRIPTION
**Pull request by mistydemeo**
Fixes https://bugs.python.org/issue37285.

https://bugs.python.org/issue37285

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Our records indicate we have not received your CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

File: setup.py [hunk 1]
```diff
@@ -55,13 +55,13 @@ def macosx_sdk_root():
...
 
-def is_macosx_sdk_path(path):
+def is_macosx_sdk_path(path, sysroot=macosx_sdk_root()):
     """
     Returns True if 'path' can be located in an OSX SDK
     """
     return ( (path.startswith('/usr/') and not path.startswith('/usr/local'))
                 or path.startswith('/System/')
-                or path.startswith('/Library/') )
+                or (path.startswith('/Library/') and not path.startswith(sysroot)) )
```
Comment (ronaldoussoren) [inline setup.py:64]: It is not immediately clear why "and not ..." is necessary. Please add a comment that explains this.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]
The same code is present in the 3.x branches as well, and should therefore be fixed there as well.

**Comment by ned-deily** [comment]
Thanks for the PR!  As explained on the bug tracker issue, the problems seen that your PR corrects have been fixed in PR #14256 and PR #14257.  These fixes are released in 2.7.17.

**Pull request closed without merging**